### DISCUSSION TITLE
Is it possible to localize DataListItem Name and Description?

### DISCUSSION THREAD

**Discussion by Nikkelmann** [leekelleher/umbraco-contentment#304] [Question] [answered]
In the backoffice it is possible to localize properties by prepending the localization key with a hash sign, eg. "#area_key".
However, this does not seem to work with the DataListItems, nor does the "area/key" format (https://docs.umbraco.com/umbraco-cms/extending/language-files#using-the-language-keys).

Is there a way to get the DataListItem Name and Description values localized?

### The setup I've tried so far

The property on the doctype with the added localization keys:
![image](https://user-images.githubusercontent.com/3674690/217805053-9c71708f-41fd-4048-8f27-0e22d6ea0b79.png)

The rendered form in the backoffice, where the property is translated, but not the names of the values:
![image](https://user-images.githubusercontent.com/3674690/217804873-acf8852d-56eb-4d6f-9fe4-040a36503487.png)

The enum with the DataListItem, where I've tried with three different localization key formats:
![image](https://user-images.githubusercontent.com/3674690/217805351-91deb41b-8e84-44b2-ab21-a166548f172f.png)
Note: I've also tried with @ in front of the key.

The DataType setup:
![image](https://user-images.githubusercontent.com/3674690/217809088-f5d1fb96-7778-4916-aec0-044c178ceceb.png)

The translations in `/config/lang/{language}.user.xml`:
![image](https://user-images.githubusercontent.com/3674690/217805577-553d020c-187b-497a-8077-590e5065ad44.png)

**Comment by leekelleher** [comment]
Hi @Nikkelmann.

There's nothing in the Data List code that will detect and localize the keys. The reason being that each of the data-sources are responsible for their own data-items. With the Enum data-source, the optional `DataListItemAttribute` is very basic, using the literal strings.

Currently, the potential options are to either roll-your-own custom data-source that injects Umbraco's `ILocalizedTextService`... or to hook into the `SendingContentNotification` handler to localize the labels (again using the `ILocalizedTextService`).

Now since you've raised this question, I would be open to adding this feature to the Enum data-source, I'm thinking along the lines of introducing some kind of `UmbracoLocalizeAttribute` (for want of a better name) - as opposed to overloading `DataListItemAttribute` with scenarios.  All things for me to consider. 🤔

**Comment by leekelleher** [comment, marked as answer]
Closing this off, as with Contentment v6.x, the Data List editors will run the item's name/label through the localization controller (client-side).
